ui/scripts/network.js

```diff
@@ -87,12 +87,21 @@
 
         } else { //non-portable IP which has only one NIC
             // Get NIC IPs
+            var data = {
+                virtualmachineid: instance.id,
+            };
+            if (args.context.networkid != undefined) {
+                $.extend(data, {
+                    networkid: args.context.networkid
+                });
+            } else {
+                $.extend(data, {
+                    networkid: network.id
+                });
+            }
```

Comment:
very minor suggestion- Alternate way of writing could be (just to make it smaller)-
```
var data = {
         virtualmachineid: instance.id,
         networkid: (args.context.networkid != undefined) ? args.context.networkid : network.id
}
```

Reply:
@anuragaw 
Changed the code as you suggested, tested ok.
thanks for good suggestion.